lib/src/lifecycle_module.dart

```diff
@@ -147,24 +189,41 @@ abstract class LifecycleModule extends Object with Disposable {
   /// Calls the onLoad() method, which can be implemented on a Module.
   /// Executes the willLoad and didLoad event streams.
   ///
-  /// [LifecycleModule] only supports one load/unload cycle. If [load] is called
-  /// again after a module has been unloaded, a [StateError] is thrown.
+  /// Initiates the loading process when the module is in the initialized state.
+  /// If the module is in the loaded or loading state a warning is logged and
+  /// the method is a noop. If the module is in any other state, a StateError is
+  /// thrown.
+  ///
+  /// Note that [LifecycleModule] only supports one load/unload cycle. If [load]
+  /// is called after a module has been unloaded, a [StateError] is thrown.
   Future<Null> load() {
-    final completer = new Completer<Null>();
-    if (_isUnloadedOrUnloading) {
-      completer
-          .completeError(new StateError('Module "$name" cannot be reloaded.'));
-    } else if (!_isLoaded) {
-      _willLoadController.add(this);
-      onLoad().then((_) {
-        _didLoadController.add(this);
-        _isLoaded = true;
-        completer.complete();
-      });
-    } else {
+    if (isLoading) {
+      _logger.warning('Module "$name" is already loading, cannot load.');
+      return _transitionFuture;
+    }
+
+    if (isLoaded) {
       _logger.warning('Module "$name" is already loaded, cannot load.');
-      completer.complete();
+      return new Future.value(null);
+    }
+
+    if (_state != LifecycleState.initialized) {
+      return new Future.error(
+          new StateError('Module "$name" state is $_state, cannot load.'));
     }
+
+    _state = LifecycleState.loading;
+    final completer = new Completer<Null>();
+    _transitionFuture = completer.future;
+    _willLoadController.add(this);
+
+    onLoad().then((_) {
+      _didLoadController.add(this);
+      _state = LifecycleState.loaded;
```

Comment:
In practice, I'm not sure this would ever make a difference since the stream is async, but it feels like the `_state` should be updated prior to emitting the event.